lib/scraper.js

```diff
@@ -127,9 +187,24 @@ Scraper.prototype.prepare = function prepare () {
 
 Scraper.prototype.load = function load () {
 	var self = this;
-	return Promise.map(self.originalResources, function loadResource (resource) {
-		return self.loadResource(resource);
+	_.forEach(self.originalResources, function loadResource (resource) {
+		self.loadResource(resource);
 	});
+	return self.waitForLoad();
+};
+
+Scraper.prototype.waitForLoad = function waitForLoad (previousCount) {
```

Comment:
I don't know how. The problem is that we need a mechanism to wait for all the resources to be loaded. Since it is a graph instead of a tree, it is not possible to have a recursively chained based waiting system.
This mechanism is actually quite simple:

1. Wait for all currently loading resources to finish.
2. If there are now new recources being loaded, go to 1. Else we're done.